.github/workflows/pr-validate-changesets.yaml

```diff
@@ -0,0 +1,145 @@
+name: Validate Changesets
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+permissions:
+  pull-requests: write
+  repository-projects: write
+
+jobs:
+  create-changeset:
+    name: Create Changeset
+    runs-on: ubuntu-latest
+    if: github.actor  == 'dependabot[bot]'
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.client_payload.ref }}
+
+      - name: Setup PNPM
+        uses: pnpm/action-setup@v4
+        with:
+          version: 9.0.5
+          run_install: true
```

Comment:
The script relies on `@changesets/cli` to generate the changeset unfortunately, but you make a good point, I think this could be optimized to avoid installing all dependencies in the future.